astronomer/providers/apache/livy/hooks/livy.py

```diff
@@ -0,0 +1,426 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""This module contains the Apache Livy hook async."""
+import asyncio
+import re
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+import aiohttp
+from aiohttp import ClientResponseError
+from airflow.providers.apache.livy.hooks.livy import BatchState
+from airflow.utils.log.logging_mixin import LoggingMixin
+from asgiref.sync import sync_to_async
+
+from astronomer.providers.http.hooks.http import HttpHookAsync
+
+
+class LivyHookAsync(HttpHookAsync, LoggingMixin):
+    """
+    Hook for Apache Livy through the REST API using LivyHookAsync
+
+    :param livy_conn_id: reference to a pre-defined Livy Connection.
+    :param extra_options: A dictionary of options passed to Livy.
```

Comment:
Would be good to add some examples of what kind of options can be passed

Reply:
[Done](https://github.com/astronomer/astronomer-providers/pull/178/commits/04c8937d588dcf893f1c2a16692aef33d8d3badd)